substrate/frame/support/src/traits/tokens/fungible/mod.rs

```diff
@@ -58,13 +58,17 @@
 //!   3 holds for 100 units, the account can spend its funds for any reason down to 300 units, at
 //!   which point the holds will start to come into play.
 //!
-//! - **Frozen Balance**: A freeze on a specified amount of an account's free balance until a
-//!   specified block number.
+//! - **Frozen Balance**: A freeze on a specified amount of an account's balance. Tokens that 
+//!   are frozen cannot be transferred.
 //!
 //!   Multiple freezes always operate over the same funds, so they "overlay" rather than
 //!   "stack". This means that if an account has 3 freezes for 100 units, the account can spend its
 //!   funds for any reason down to 100 units, at which point the freezes will start to come into
 //!   play.
+//!   
+//!   It's important to note that the frozen balance can exceed the total balance of the account.
+//!   In such cases the excess frozen amount doesn't actually belong to the account, only the lock
+//!   does.
```

Comment:
```suggestion
//!   It's important to note that the frozen balance can exceed the total balance of the account.
//!   This can be used in scenarios where you would want to keep an account's balance frozen, independent of the specific amount of balance they have. For example, by setting the users frozen balance to `Balance::MAX`, the user will not be able to transfer any funds in the account. Developers should be conscious to use the `balance_freezable` function to determine the amount of available balance of the user which can become frozen.
```

Here is some un-formatted word-vomit.

Perhaps give it a read yourself and write it in your own words which make sense to a new user.

Reply:
Awesome, I've to leave now but I'll rephrase it asap. Having an use example is helpful cause the idea isn't natural at all 👍

Reply:
> Here is some un-formatted word-vomit.
> 
> Perhaps give it a read yourself and write it in your own words which make sense to a new user.

I didn't include finally the mention to the `balance_freezable` function as I think it may lead to confusion anyway. The amount specified in that function may be bypassed using `Fortitude::Force`, so saying in the same sentence that there's no upper bound in the freezable amount and that developers may use this function to set such a bound feels weird IMO. Anyway, the purpose of that function's well explained in its own doc

Reply:
all good to me